### DISCUSSION TITLE
A self evolving Darwin?

### DISCUSSION THREAD

**Discussion by sdmorrey** [Cognation/darwin#9] [Ideas]
The idea is pretty obvious. Would it be possible for Darwin to improve itself?

First improvement would be have it run entirely locally.
This could be done by compiling llama.cpp or equivalent and choosing a good code generating model such as the recent Granite series by IBM. It could then modify its own code to connect to that.

From there I would have a job that monitors system activity and while the system is idle have it go over its own code looking for optimizations and refinements as well as possible enhancements. Ideally it would work on a copy of itself perhaps with a supervisory daemon that could restart and/or revert and restart if it gets into trouble.

Just some thoughts, because any good software engineer is always looking for ways to improve.

**Comment by 9throok** [comment]
@sdmorrey sounds pretty interesting. I can look into this and see how it pans out. The idea of running it entirely locally is surely in the pipeline, but right now we are more focused on improving the performance of Darwin.

And ofcourse, your ideas are always welcome.
